examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java

```diff
@@ -258,6 +263,108 @@ void createOtherGroup () {
 		resizableColumns.addSelectionListener (widgetSelectedAdapter(event -> setColumnsResizable ()));
 		headerImagesButton.addSelectionListener (widgetSelectedAdapter(event -> recreateExampleWidgets ()));
 		subImagesButton.addSelectionListener (widgetSelectedAdapter(event -> recreateExampleWidgets ()));
+		editableButton.addSelectionListener(widgetSelectedAdapter(event -> makeTreeContentEditable()));
+	}
+
+	/**
+	 * Make Tree Editable
+	 */
+	void makeTreeContentEditable() {
+		// Tree 1
+		final TreeEditor editor = new TreeEditor(tree1);
+		editor.horizontalAlignment = SWT.LEFT;
+		editor.grabHorizontal = true;
+		// Add double click listener to the tree
+		tree1.addListener(SWT.MouseDoubleClick, new Listener() {
+			@Override
+			public void handleEvent(Event event) {
+				if (editableButton.getSelection()) {
+					Point point = new Point(event.x, event.y);
+					TreeItem item = tree1.getItem(point);
+					if (item != null) {
+						// Get the item text
+						final String oldText = item.getText();
+
+						// Create a text field to edit the item text
+						final Text text = new Text(tree1, SWT.NONE);
+						text.setText(oldText);
+						text.selectAll();
+						text.setFocus();
+
+						// Add a focus out listener to commit changes on focus lost
+						text.addListener(SWT.FocusOut, new Listener() {
```

Comment:
Like above: `Listener` is a functional interface, so you can use a lambda instead of providing an anonymous implementation of that interface.